diff --git a/protocol/openthread/component/ot_platform_abstraction.slcc b/protocol/openthread/component/ot_platform_abstraction.slcc
index ba707d733..9ee266d11 100644
--- a/protocol/openthread/component/ot_platform_abstraction.slcc
+++ b/protocol/openthread/component/ot_platform_abstraction.slcc
@@ -31,6 +31,7 @@ config_file:
   - path: protocol/openthread/config/sl_openthread_rtos_config.h
     condition:
       - kernel
+    unless: ["matter_integration"]
   - path: protocol/openthread/config/sl_openthread_subghz_config.h
     condition:
       - hardware_board_supports_ot_subghz
@@ -39,12 +40,14 @@ toolchain_settings:
     value: sl_openthread_rtos_config.h
     condition:
       - kernel
+    unless: ["matter_integration"]
   - option: preinclude
     value: sl_openthread_subghz_config.h
     condition:
       - hardware_board_supports_ot_subghz
 include:
   - path: protocol/openthread/platform-abstraction/rtos
+    unless: ["matter_integration"]
     file_list:
       - path: sl_ot_rtos_adaptation.h
         condition:
@@ -53,6 +56,7 @@ source:
   - path: protocol/openthread/platform-abstraction/rtos/sl_ot_rtos_adaptation.c
     condition:
       - kernel
+    unless: ["matter_integration"]
 define:
   - name: RADIO_CONFIG_DMP_SUPPORT
     value: 1
@@ -62,6 +66,7 @@ define:
     value: 1
     condition:
       - freertos
+    unless: ["matter_integration"]
   - name: OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE
     value: 1
     condition:
@@ -78,6 +83,7 @@ template_contribution:
       handler: sl_ot_rtos_init
     condition:
       - kernel
+    unless: ["matter_integration"]
   - name: ot_init
     value: sl_ot_create_instance
     priority: 0
diff --git a/protocol/openthread/component/ot_platform_abstraction_core.slcc b/protocol/openthread/component/ot_platform_abstraction_core.slcc
index b71fb6dc1..6442d0710 100644
--- a/protocol/openthread/component/ot_platform_abstraction_core.slcc
+++ b/protocol/openthread/component/ot_platform_abstraction_core.slcc
@@ -55,9 +55,11 @@ source:
       - path: protocol/openthread/platform-abstraction/efr32/uartdrv_uart.c
         condition:
           - uartdrv_usart
+        unless: ["matter_integration"]
       - path: protocol/openthread/platform-abstraction/efr32/iostream_uart.c
         condition:
           - iostream_usart
+        unless: ["matter_integration"]
       - path: protocol/openthread/platform-abstraction/efr32/sleep.c
         condition:
           - power_manager
diff --git a/protocol/openthread/component/ot_platform_abstraction_source.slcc b/protocol/openthread/component/ot_platform_abstraction_source.slcc
index 6e42c698a..774cc9c13 100644
--- a/protocol/openthread/component/ot_platform_abstraction_source.slcc
+++ b/protocol/openthread/component/ot_platform_abstraction_source.slcc
@@ -38,7 +38,9 @@ source:
       - path: protocol/openthread/platform-abstraction/efr32/entropy.c
       - path: protocol/openthread/platform-abstraction/efr32/flash.c
       - path: protocol/openthread/platform-abstraction/efr32/logging.c
+        unless: ["matter_integration"]
       - path: protocol/openthread/platform-abstraction/efr32/memory.c
+        unless: ["matter_integration"]
       - path: protocol/openthread/platform-abstraction/efr32/misc.c
       - path: protocol/openthread/platform-abstraction/efr32/crypto.c
       - path: protocol/openthread/platform-abstraction/efr32/ieee802154-packet-utils.cpp
diff --git a/util/third_party/crypto/sl_component/sl_mbedtls_support/config/template/mbedtls_config_autogen.h.jinja b/util/third_party/crypto/sl_component/sl_mbedtls_support/config/template/mbedtls_config_autogen.h.jinja
index 321418382..80317974a 100644
--- a/util/third_party/crypto/sl_component/sl_mbedtls_support/config/template/mbedtls_config_autogen.h.jinja
+++ b/util/third_party/crypto/sl_component/sl_mbedtls_support/config/template/mbedtls_config_autogen.h.jinja
@@ -28,7 +28,12 @@
 
 #if !defined(TEST_SUITE_MEMORY_BUFFER_ALLOC)
 #if !defined(MBEDTLS_PLATFORM_FREE_MACRO) && !defined(MBEDTLS_PLATFORM_CALLOC_MACRO)
-#if defined(CONFIG_MBEDTLS_USE_FREERTOS_PVCALLOC)
+#if defined(CONFIG_MBEDTLS_USE_CHIP_PVCALLOC)
+    #include <lib/support/CHIPPlatformMemory.h>
+    #define MBEDTLS_PLATFORM_CALLOC_MACRO CHIPPlatformMemoryCalloc
+    #define MBEDTLS_PLATFORM_FREE_MACRO   CHIPPlatformMemoryFree
+
+#elif defined(CONFIG_MBEDTLS_USE_FREERTOS_PVCALLOC)
     /* In FreeRTOS, use pvCalloc (and vPortFree) for dynamic memory allocation.
        E.g. Amazon FreeRTOS implements pvCalloc for dynamic memory allocation. */
     #include <stddef.h>
